### PR TITLE
[project-base] redesigned homepage

### PR DESCRIPTION
**Pull request by chlebektomas**
#### Description, the reason for the PR
Homepage has a new look based on design.

#### Fixes issues
... 

#### Check the appropriate checkboxes below, please

- [ ] [BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/) 
- [ ] New feature 
- [x] I have read and signed [License Agreement for contributions](https://www.shopsys.com/license-agreement)

----
:globe_with_meridians: Live Preview:
  - https://tc-ssp-2669-redesign-homepage.odin.shopsys.cloud
  - https://cz.tc-ssp-2669-redesign-homepage.odin.shopsys.cloud

### CODE REVIEW

File: project-base/storefront/components/Blocks/Banners/BannersSlider.tsx [hunk 1]
```diff
@@ -66,33 +66,39 @@ export const BannersSlider: FC<BannersSliderProps> = ({ sliderItems }) => {
...
+        "vl:after:sticky vl:after:right-0 vl:after:top-0 vl:after:block vl:after:h-full vl:after:w-3 vl:after:bg-backgroundDark vl:after:bg-gradient-to-r vl:after:from-background vl:after:to-transparent vl:after:opacity-25 vl:after:content-['']",
+    );
+
     return (
-        <div className="flex flex-col gap-6 vl:flex-row" tid={TIDs.banners_slider}>
+        <div className="flex flex-col" tid={TIDs.banners_slider}>
             <div
                 {...handlers}
-                className="rounded vl:basis-3/4"
+                className=""
```
Comment (JanMolcik) [inline project-base/storefront/components/Blocks/Banners/BannersSlider.tsx]: Please remove empty className.

File: project-base/storefront/components/Blocks/BlogPreview/BlogPreview.tsx [hunk 2]
```diff
@@ -18,45 +20,47 @@ export type BlogPreviewProps = {
...
-    const blogSideItems = useMemo(
-        () => mapConnectionEdges<TypeListedBlogArticleFragment>(blogArticles?.slice(2)),
-        [blogArticles],
-    );
+    const blogItems = useMemo(() => mapConnectionEdges<TypeListedBlogArticleFragment>(blogArticles), [blogArticles]);
+    const blogMainItems = blogItems?.slice(0, 2);
+    const blogSideItems = blogItems?.slice(2);
+
+    const { width } = useGetWindowSize();
+    const isDesktop = width > desktopFirstSizes.notLargeDesktop;
```
Comment (JanMolcik) [inline project-base/storefront/components/Blocks/BlogPreview/BlogPreview.tsx]: I think you can achieve this easier with example from `Pagination.tsx` file - 
```typescript
const isDesktop = useMediaMin('vl');
```

File: project-base/storefront/components/Blocks/BlogPreview/BlogPreviewSide.tsx [hunk 3]
```diff
@@ -4,43 +4,53 @@ import { Image } from 'components/Basic/Image/Image';
...
-            </ArticleLink>
-        ))}
-    </div>
-);
+                    <div className="flex flex-col items-start gap-2">
+                        <div className="flex items-center gap-6 whitespace-nowrap">
+                            <span className="text-inputPlaceholder text-sm font-secondary font-semibold">
+                                {formatDate(article.publishDate, 'l')}
+                            </span>
+                            {article.blogCategories.map((blogPreviewCategorie) => (
```
Comment (JanMolcik) [inline project-base/storefront/components/Blocks/BlogPreview/BlogPreviewSide.tsx]: Please fix the name to `blogPreviewCategory`.

File: project-base/storefront/components/Blocks/BlogPreview/BlogPreviewSide.tsx [hunk 4]
```diff
@@ -4,43 +4,53 @@ import { Image } from 'components/Basic/Image/Image';
...
+                                <Fragment key={blogPreviewCategorie.uuid}>
+                                    {blogPreviewCategorie.parent && (
+                                        <Flag href={blogPreviewCategorie.link} type="blog">
+                                            {blogPreviewCategorie.name}
+                                        </Flag>
+                                    )}
+                                </Fragment>
```
Comment (JanMolcik) [inline project-base/storefront/components/Blocks/BlogPreview/BlogPreviewSide.tsx]: Do we need the `<Fragment>` here or could we just provide `key` straight to the `<Flag ... key={blogPreviewCategory.uuid}>` component?

File: project-base/storefront/components/Blocks/Categories/PromotedCategories.tsx [hunk 5]
```diff
@@ -1,17 +1,25 @@
...
+    return (
+        <Webline className="mb-10">
+            <h3 className="mb-4">{t('Shop by category')}</h3>
+            {arePromotedCategoriesFetching ? (
+                <SkeletonModulePromotedCategories />
+            ) : (
+                <PromotedCategoriesContent promotedCategoriesData={promotedCategoriesData} />
+            )}
+        </Webline>
+    );
```
Comment (JanMolcik) [inline project-base/storefront/components/Blocks/Categories/PromotedCategories.tsx]: Are you sure this still shows the skeleton loader?

File: project-base/storefront/utils/useEffectOnce.ts
Comment (JanMolcik) [inline project-base/storefront/utils/useEffectOnce.ts:1]: Do we really need this utility?

File: project-base/storefront/components/Blocks/Product/DeferredPromotedProducts.tsx [hunk 6]
```diff
@@ -16,24 +18,32 @@ const ProductsSliderPlaceholder = dynamic(() =>
...
+    const { t } = useTranslation();
     const [{ data: promotedProductsData, fetching: arePromotedProductsFetching }] = usePromotedProductsQuery();
     const shouldRender = useDeferredRender('promoted_products');
 
-    if (arePromotedProductsFetching) {
-        return <SkeletonModulePromotedProducts />;
-    }
-
-    if (!promotedProductsData?.promotedProducts) {
+    if (!promotedProductsData?.promotedProducts.length) {
```
Comment (JanMolcik) [inline project-base/storefront/components/Blocks/Product/DeferredPromotedProducts.tsx]: Same question, does the skeleton loader works with this change?

**Review by JanMolcik — CHANGES_REQUESTED** [review]

File: project-base/storefront/components/Blocks/Banners/BannersSlider.tsx (hunk 1 above)
Comment (chlebektomas) [reply project-base/storefront/components/Blocks/Banners/BannersSlider.tsx]: fixed

File: project-base/storefront/components/Blocks/BlogPreview/BlogPreview.tsx (hunk 2 above)
Comment (chlebektomas) [reply project-base/storefront/components/Blocks/BlogPreview/BlogPreview.tsx]: Nice, thanks for that!

File: project-base/storefront/components/Blocks/BlogPreview/BlogPreviewSide.tsx (hunk 3 above)
Comment (chlebektomas) [reply project-base/storefront/components/Blocks/BlogPreview/BlogPreviewSide.tsx]: fixed

File: project-base/storefront/utils/useEffectOnce.ts
Comment (chlebektomas) [reply project-base/storefront/utils/useEffectOnce.ts:1]: removed

File: project-base/storefront/components/Blocks/BlogPreview/BlogPreviewSide.tsx (hunk 4 above)
Comment (chlebektomas) [reply project-base/storefront/components/Blocks/BlogPreview/BlogPreviewSide.tsx]: fixed

File: project-base/storefront/components/Blocks/Categories/PromotedCategories.tsx (hunk 5 above)
Comment (chlebektomas) [reply project-base/storefront/components/Blocks/Categories/PromotedCategories.tsx]: fixed

File: project-base/storefront/components/Blocks/Product/DeferredPromotedProducts.tsx (hunk 6 above)
Comment (chlebektomas) [reply project-base/storefront/components/Blocks/Product/DeferredPromotedProducts.tsx]: fixed

**Review by JanMolcik — APPROVED** [review]

**Review by JanMolcik — APPROVED** [review]

**Pull request merged**